pallets/fragments/src/lib.rs

```diff
@@ -166,9 +166,9 @@ pub enum Currency<TFungibleAsset> {
 
 /// **Struct** of a **Fragment Definition's Metadata**
 #[derive(Encode, Decode, Clone, scale_info::TypeInfo, Debug, PartialEq)]
-pub struct DefinitionMetadata<TFungibleAsset> {
+pub struct DefinitionMetadata<TU8Vector, TFungibleAsset> {
```

Comment:
Why?

Reply:
You do not need to modify the structs fields to be `BoundedVec`. Only having them as inputs in the extrinsics, as you already did btw, is enough. It's just a check on the inputs from the caller, you can keep them as `Vec<u8>` for storage, so consequently in the structs.

Reply:
> Why?

Because `fragments.create()` has a parameter of type `DefinitionMetadata`. Therefore, if we don't bound that parameter, a definition's name can be of any length!

Reply:
I bounded the parameter in `fragments.create()` rather than using `ensure!()` because then will even prevent the extrinsic from entering into the block if its `DefinitionMetadata` parameter is too long. so this prevents polluting/spamming of blocks

Reply:
yeah ok